### DISCUSSION TITLE
Does chakra-ui support RTL?

### DISCUSSION THREAD

**Discussion by nalhusail** [chakra-ui/chakra-ui#2144] [Q&A] [answered]
Hi all,
I'm evaluating ui frameworks for react, and chakra is amazing

I'm just wondering if chakra-ui supports RTL out of the box, it could be a setting I activate, or a few lines of code I add

What I consider as RTL friendly is if I create the layout in LTR, then switch this setting on, then everything flips correctly, without me having to redo the UI for each page

I couldn't find an answer for this question in the context of chakra-ui, but I did find it in relation to styled-system here
https://medium.com/styled-components/announcing-styled-components-v5-beast-mode-389747abd987

assuming chakra doesn't have this property, is it possible to use the StyleSheetManager as an alternative solution?

**Comment by ljosberinn** [comment, marked as answer]
we're unaware of any incompatibilities with RTL - all you'd need to do is to place the correct `dir` attribute on `html`, the rest would be dev layouting

we've had a few people asking about it and here is a related issue: https://github.com/chakra-ui/chakra-ui/issues/517#issuecomment-668257818

**Comment by chasinhues** [comment]
Marking this discussion as resolved due to inactivity. If anyone else lands here trying to find out about Chakra's RTL Support, please check out https://chakra-ui.com/docs/features/rtl-support 🎉
